app/src/main/java/dcom/taneja/dhruv/floater/service/FloaterService.java

```diff
@@ -0,0 +1,135 @@
+package dcom.taneja.dhruv.floater.service;
+
+import android.app.Service;
+import android.content.Intent;
+import android.graphics.PixelFormat;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+
+import dcom.taneja.dhruv.floater.R;
+
+/**
+ * Created by dhruvtaneja on 10/02/18.
+ */
+
+public class FloaterService extends Service {
+
+    private View mFloatingView;
+    private View mExpandedView;
+    private WindowManager mWindowManager;
+    private WindowManager.LayoutParams mLayoutParams;
+
+    private int mInitialX;
+    private int mInitialY;
+    private int mInitialTouchX;
+    private int mInitialTouchY;
+    private boolean mIsCollapsed = true;
+    private boolean mHasMoved;
+    private boolean mDragCollapse;
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        mFloatingView = LayoutInflater.from(this).inflate(R.layout.layout_floating_widget,
+                null);
+        mLayoutParams = new WindowManager.LayoutParams(
+                WindowManager.LayoutParams.MATCH_PARENT,
+                WindowManager.LayoutParams.WRAP_CONTENT,
+                WindowManager.LayoutParams.TYPE_PHONE,
+                WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE,
+                PixelFormat.TRANSLUCENT
+        );
+
+        mLayoutParams.gravity = Gravity.TOP | Gravity.END;
+        mLayoutParams.x = 0;
+        mLayoutParams.y = 100;
+        mLayoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;
+
+        mWindowManager = (WindowManager) getSystemService(WINDOW_SERVICE);
+        mWindowManager.addView(mFloatingView, mLayoutParams);
```

Comment:
Could you provide some comments on the code? Having a hard time understanding this right now.
Not a requirement, can be done later. Will approve it after testing it a little on my environment.